### PR TITLE
*: show create table output utf8_bin collate.

### PR DESCRIPTION
**Pull request by coocood**
We only support case sensitive utf8_bin collation right now, but show create table ignored the collation.
So the dumped DDL statement doesn't contains collation option.
When the DDL statement is executed on MySQL, the collation will be changed to utf8_general_ci.
If the string column is defined as unique key, there would be duplicated key error.

This PR enforce utf8 charset and utf8_bin collation when parse a column definition, and enfore table option
`DEFAULT CHARSET=utf8 COLLATE=utf8_bin` in SHOW CREATE TABLE statement.

### CODE REVIEW

**Comment by coocood** [comment]
@shenli @tiancaiamao @zimulala @hanfei1991 
PTAL

File: parser/parser.y [hunk 1]
```diff
@@ -5933,36 +5926,38 @@ StringType:
...
 	}
 |	TextType OptBinary OptCharset OptCollate
 	{
 		x := $1.(*types.FieldType)
-		if $2.(bool) {
-			x.Flag |= mysql.BinaryFlag
-		}
-		x.Charset = $3.(string)
-		x.Collate = $4.(string)
+		x.Charset = charset.CharsetUTF8
```
Comment (shenli) [inline parser/parser.y]: Why should discuss about this.

**Comment by coocood** [comment]
@shenli PTAL

File: parser/parser_test.go [hunk 2]
```diff
@@ -1548,3 +1549,38 @@ func (s *testParserSuite) TestSQLModeANSIQuotes(c *C) {
 		c.Assert(err, IsNil)
 	}
 }
+
+func (s *testParserSuite) TestDDLStatements(c *C) {
+	parser := New()
+	parser.SetSQLMode(mysql.ModeANSIQuotes)
```
Comment (tiancaiamao) [inline parser/parser_test.go]: why need we set this?

**Comment by shenli** [comment]
LGTM

File: parser/parser_test.go [hunk 3]
```diff
@@ -1548,3 +1549,38 @@ func (s *testParserSuite) TestSQLModeANSIQuotes(c *C) {
...
+			c.Assert(tblOpt.StrValue, Equals, "utf8")
+		case ast.TableOptionCollate:
+			c.Assert(tblOpt.StrValue, Equals, "utf8_bin")
+		}
+	}
+	createTableStr = `CREATE TABLE t (
+		a varbinary(64),
+		b binary(10),
+		c blob)`
+	stmts, err = parser.Parse(createTableStr, "", "")
```
Comment (zimulala) [inline parser/parser_test.go]: Add error check.

File: parser/parser.y [hunk 4]
```diff
@@ -4176,9 +4176,6 @@ CastType:
 	{
 		x := types.NewFieldType(mysql.TypeString)
 		x.Flen = $2.(int)
-		if $3.(bool) {
-			x.Flag |= mysql.BinaryFlag
-		}
 		x.Charset = $4.(string)
```
Comment (zimulala) [inline parser/parser.y]: Need to set `collate` ?

Comment (coocood) [inline parser/parser.y]: The collate is not given by the syntax, the default collation will be used in runtime.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**